lib/Runtime/Library/TypedArray.cpp

```diff
@@ -143,7 +143,11 @@ namespace Js
             {
                 if (JavascriptOperators::IsObject(firstArgument))
                 {
-                    if (JavascriptOperators::IsIterable(RecyclableObject::FromVar(firstArgument), scriptContext))
+                    if (JavascriptOperators::IsIterable(RecyclableObject::FromVar(firstArgument), scriptContext) &&
+                        (!JavascriptArray::Is(firstArgument) ||
+                        JavascriptOperators::HasOwnProperty(firstArgument, PropertyIds::_symbolIterator, scriptContext) ||
+                        DynamicObject::FromVar(firstArgument)->GetPrototype() != scriptContext->GetLibrary()->GetArrayPrototype() ||
+                        scriptContext->GetLibrary()->GetArrayObjectHasUserDefinedIterator()))
```

Comment:
`GetProperty` can call script code, on a getter property for instance, but particularly on a proxy object where property get can be trapped.  `GetProperty` and `SetProperty` can call script code and this something important to consider when using them.

Using `IsIterable` anywhere is a bug because of this.  I've opened issue #584 to track this separately.